crates/tools/src/remote_block.rs

```diff
@@ -1,9 +1,62 @@
-use edr_eth::chain_spec::L1ChainSpec;
+use core::fmt::Debug;
+
+use clap::ValueEnum;
+use edr_eth::{
+    chain_spec::L1ChainSpec, log::FilterLog, result::InvalidTransaction,
+    transaction::TransactionValidation,
+};
 use edr_evm::test_utils::run_full_block;
+use edr_optimism::OptimismChainSpec;
+use edr_provider::spec::SyncChainSpec;
 use edr_rpc_eth::client::EthRpcClient;
 
-pub async fn replay(url: String, block_number: Option<u64>, chain_id: u64) -> anyhow::Result<()> {
-    let rpc_client = EthRpcClient::<L1ChainSpec>::new(&url, edr_defaults::CACHE_DIR.into(), None)?;
+#[derive(Clone, ValueEnum)]
+pub enum SupportedChainTypes {
+    L1,
+    Optimism,
+}
+
+pub async fn replay(
+    chain_type: SupportedChainTypes,
+    url: String,
+    block_number: Option<u64>,
+    chain_id: u64,
```

Comment:
I know this was already here, but I hadn't noticed it before. Do we need this parameter?

`edr_evm::test_utils::run_full_block` could uses the `chain_id` after it created the `rpc_client`, so it could be just fetched.

`chain_type`, on the other hand, seems like it can't be avoided, or at least it shouldn't.

Reply:
That's a very good question!

In other places we explicitly override the chain ID with a local chain ID - for security reasons. Here, I don't think that's necessary.

Reply:
Addressed in https://github.com/NomicFoundation/edr/pull/638/commits/5805442fae323eb07bfd7aaa0f51044dde199a6e